### PR TITLE
Add helperText prop to ArrayInput

### LINKED ISSUE #4227: ArrayInput does not support helperText prop
**What you were expecting:**
As ArrayInput should accept the common input props, helperText should be working, too.

**What happened instead:**
HelperText prop is not handled in the input, therefore does not appear.

**Related code:**
I'd love to create a pull request, however, I am not sure how the related files are compiled. Making the prop work would only need a few lines of code into ArrayInput.tsx:

```
import {  InputLabel, FormControl, FormHelperText } from '@material-ui/core';
...
<InputLabel ... />
<FormHelperText>{helperText}</FormHelperText>
```
Besides this, adding helperText const to the props and the proptypes.

**Environment**

* React-admin version: 3.1.1

### PR DESCRIPTION
**Pull request by netdown**
Fixes #4227

(port of #4236 in next branch)

### CODE REVIEW

**Comment by fzaninotto** [comment]
Tests fail because of https://github.com/final-form/react-final-form-arrays/issues/78. We cant merge this one until react-final-form-arrays fixes their bug upstream.

File: packages/ra-ui-materialui/src/input/ArrayInput.tsx [hunk 1]
```diff
@@ -70,21 +72,37 @@ export const ArrayInput = ({
...
+                error={touched && !!error}
+            >
                 <FieldTitle
                     label={label}
                     source={source}
                     resource={resource}
                     isRequired={isRequired(validate)}
                 />
             </InputLabel>
+            {(touched && error) || helperText ? (
```
Comment (archfz) [inline packages/ra-ui-materialui/src/input/ArrayInput.tsx:97]: For consistency shouldn't this be under the items? For other inputs we don't put the helper text between the label and the form control. @fzaninotto what do you think ?

**Comment by netdown** [comment]
@fzaninotto What is the reason of closing?

**Comment by WiXSL** [comment]
> Tests fail because of [final-form/react-final-form-arrays#78](https://github.com/final-form/react-final-form-arrays/issues/78). We cant merge this one until react-final-form-arrays fixes their bug upstream.

This has been fixed in version [3.1.2](https://github.com/final-form/react-final-form-arrays/releases/tag/v3.1.2)

**Pull request closed without merging**